ml/main.py

```diff
@@ -0,0 +1,152 @@
+
+
+def add_calorie_labels(calorie_data, low_calorie_threshold,
+                       high_calorie_threshold):
+  """Set 'calorie_label' for each row in the dataset to reflect which class of calorie that dish falls under"""
+
+  def label_row(calorie_data_row):
+    if calorie_data_row['amount'] < low_calorie_threshold:
+      return [1, 0, 0] # low calorie
+    elif calorie_data_row['amount'] > high_calorie_threshold:
+      return [0, 0, 1] # high calorie
+    else:
+      return [0, 1, 0] # avg calorie
+
+  calorie_data['calorie_label'] = calorie_data.apply(
+      lambda row: label_row(row), axis=1)
+
+  return calorie_data
+
+
+def num_unique_words(series):
+  """Counts the number of unique words (space-separated) in all strings of a pandas Series."""
+  words = set()
+  for entry in series:
+    words.update(entry.split(' +'))
+
+  return len(words)
+
+
+def max_sequence_length(series):
+  """Compute the max length of the values in a pandas Series."""
+  max_len = None
+  for entry in series:
+    if max_len is None or len(entry) > max_len:
+      max_len = len(entry)
+
+  return max_len
+
+def add_input_labels(calorie_data, vocab_size, max_corpus_length):
+  """Set 'input' for each row in the dataset to reflect the encoded input of the text."""
+
+  def input_for_row(row):
+    encoded = keras.preprocessing.text.one_hot(row['description'], vocab_size)
+    
+    while len(encoded) < max_corpus_length:
+      encoded.append(0)
+
+    return encoded
+
+  calorie_data['input'] = calorie_data.apply(lambda row: input_for_row(row), axis=1)
+  return calorie_data
+
+
+# %% read data from disk
+print(f'Tensorflow version {tf.__version__}')
+calorie_data = load_calorie_data()
+add_calorie_labels(calorie_data, low_calorie_threshold=100, high_calorie_threshold=500)
+
+# %% add encode 'description' into a new 'input' column in calorie_data
+calorie_data['description'] = calorie_data['description'].str.replace(',', '').str.lower()
+corpus = calorie_data['description']
+vocab_size = num_unique_words(corpus)
+tokenized_corpus = corpus.map(lambda desc: desc.split(' '))
+max_corpus_length = max_sequence_length(tokenized_corpus)
+calorie_data = add_input_labels(calorie_data, vocab_size, max_corpus_length)
+
+# %% create model
+model = keras.Sequential([
+  keras.layers.Embedding(vocab_size, int(vocab_size**(1/4)), input_length=max_corpus_length),
+  keras.layers.Flatten(),
+  keras.layers.Dense(3, activation='softmax'),
+])
+
+model.compile(
+  optimizer='adam',
+  loss='categorical_crossentropy',
+  metrics=['categorical_accuracy', keras.metrics.Precision(), keras.metrics.Recall()]
+)
+
+# %% model stats
+model.summary()
+model._layers = [layer for layer in model._layers if not isinstance(layer, dict)] # workaround for bug in keras.util.plot_model
+keras.utils.plot_model(model, show_layer_names=False, show_shapes=True)
+
+# %% split dataset
+train_frac = 0.6
+validation_frac = 0.2
+test_frac = 0.2
+
+train_set, validation_set, test_set = np.split(
```

Comment:
probably better to use scikit_learn, specifically https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html. To create these three sets, you can just run this function twice

Reply:
Fixed in https://github.com/googleinterns/amaranth/pull/2/commits/5cc2a688f69aba3abbe2f2089a7fd640eaeb82e8